source/vibe/web/rest.d

```diff
@@ -969,12 +987,13 @@ private string genClientBody(alias Func)() {
 				static if (is(PT == Json))
 					url_prefix = q{urlEncode(id.toString())~"/"};
 				else
-					url_prefix = q{urlEncode(toRestString(serializeToJson(id)))~"/"};
+					url_prefix = q{urlEncode(toRestString(id))~"/"};
 			} else static if (anySatisfy!(mixin(GenCmp!("ClientFilter", i, ParamNames[i]).Name), paramAttr)) {
 				alias paramsArgList = Filter!(mixin(GenCmp!("ClientFilter", i, ParamNames[i]).Name), UDATuple!(WebParamAttribute, Func));
-				static if (paramsArgList[0].origin == WebParamAttribute.Origin.Header)
-					param_handling_str ~= format(q{headers__["%s"] = to!string(%s);}, paramsArgList[0].field, paramsArgList[0].identifier);
-				else static if (paramsArgList[0].origin == WebParamAttribute.Origin.Query)
+				static if (paramsArgList[0].origin == WebParamAttribute.Origin.Header) {
+					static if (isNullable!PT) param_handling_str ~= format(q{ if (!%s.isNull) }, paramsArgList[0].identifier);
+					param_handling_str ~= format(q{headers__["%s"] = toRestString(%s);}, paramsArgList[0].field, paramsArgList[0].identifier);
```

Comment:
You're right, there's a possibility of breaking applications that rely on headers being serialized via a simple `to!string` call, and that's a problem.

On the other hand, should anyone really be sending non-scalar values over HTTP headers? I don't think so. But if they seriously need to do it, shouldn't it be done the same way it's done everywhere else?

Or maybe we need to introduce a general mechanism that allows users to change how different parameters are serialized.